frontend/src/components/MapView/LeftPanel/ChartsPanel/index.tsx

```diff
@@ -887,12 +751,33 @@ const ChartsPanel = memo(
             ))}
           </Select>
         </FormControl>
-        <Button
-          className={classes.downloadButton}
-          onClick={downloadCsv([
-            [dataForCsv, generateCSVFilename()],
-            [dataForSecondCsv, generateSecondCSVFilename()],
-          ])}
+        <DownloadCsvButton
+          filesData={[
+            {
+              fileName: buildCsvFileName([
+                multiCountry ? admin0Key : country,
+                selectedAdmin1Area ?? '',
+                selectedAdmin2Area ?? '',
+                ...(selectedLayerTitles as string[]),
+                comparePeriods ? 'first_period' : '',
```

Comment:
Let's do that before hand and keep telling names such as `firstCSVFilename`  `secondCSVFilename`

Reply:
alternativelt, you could create a utility function to generate the filename from params, but this bloc of logic (x2) is not great

Reply:
True haha, I meant that outside of a function called `generateCSVFilename` it feels a bit confusing.

Maybe we could simply use named arguments in buildCSVFilename and handle the `?? ''` in there.

Reply:
I think we can simply remove the `?? ''` because we already filter null, undefined, 0, empty string data in buildCSVFilename. And ok to create two constants firstCSVFilename and secondCSVFilename before calling the component. It is more readable. I do that.

Reply:
I think we can simply remove the `?? ''` because we already filter null, undefined, 0, empty string data in buildCSVFilename. And ok to create two constants firstCSVFilename and secondCSVFilename before calling the component. It is more readable. I do that.